pkg/defaults/defaults.go

```diff
@@ -120,13 +121,13 @@ func FromConfig(
 			if err != nil {
 				return nil, nil, fmt.Errorf("unable to access image stream tag on remote cluster: %v", err)
 			}
-			step = steps.SourceStep(*rawStep.SourceStepConfiguration, config.Resources, buildClient, srcClient, imageClient, artifactDir, jobSpec, dryLogger)
+			step = steps.SourceStep(*rawStep.SourceStepConfiguration, config.Resources, buildClient, srcClient, imageClient, artifactDir, jobSpec, dryLogger, sshSecretName)
 		} else if rawStep.ProjectDirectoryImageBuildStepConfiguration != nil {
 			step = steps.ProjectDirectoryImageBuildStep(*rawStep.ProjectDirectoryImageBuildStepConfiguration, config.Resources, buildClient, imageClient, imageClient, artifactDir, jobSpec, dryLogger)
 		} else if rawStep.ProjectDirectoryImageBuildInputs != nil {
-			step = steps.GitSourceStep(*rawStep.ProjectDirectoryImageBuildInputs, config.Resources, buildClient, imageClient, artifactDir, jobSpec, dryLogger)
+			step = steps.GitSourceStep(*rawStep.ProjectDirectoryImageBuildInputs, config.Resources, buildClient, imageClient, artifactDir, jobSpec, dryLogger, sshSecretName)
 		} else if rawStep.RPMImageInjectionStepConfiguration != nil {
-			step = steps.RPMImageInjectionStep(*rawStep.RPMImageInjectionStepConfiguration, config.Resources, buildClient, routeGetter, imageClient, artifactDir, jobSpec, dryLogger)
+			step = steps.RPMImageInjectionStep(*rawStep.RPMImageInjectionStepConfiguration, config.Resources, buildClient, routeGetter, imageClient, artifactDir, jobSpec, dryLogger, sshSecretName)
```

Comment:
Can `GitSourceStep` and `RPMImageInjectionStep` ever use the key?

Reply:
Yes. For example `build_root.project_image`, it will need to access a `Dockerfile` in the private repo.

Reply:
Then we need to change this as well: https://github.com/openshift/ci-tools/pull/178/files#diff-abdbc02ab75ad52d862e0625802eb296L35.

Reply:
addressed!

Reply:
We're still changing `RPMImageInjectionStep`, that one [can't possibly use the secret](https://github.com/openshift/ci-tools/blob/6902e9ce6a95261f639739672c4b3763065246ea/pkg/steps/rpm_injection.go#L15-L18).

Reply:
on second look, I don't even think that we have to add any secret to the `RPMImageInjectionStep`

Reply:
i removed the commit.